src/controllers/user.controller.js

```diff
@@ -0,0 +1,74 @@
+import ResponseService from '../services/response.service';
+import UserService from '../services/user.service';
+import BcryptService from '../services/bcrypt.service';
+import JwtService from '../services/jwt.service';
+
+
+/**
+ *
+ *
+ * @class UserController
+ */
+class UserController {
+  /**
+ * users can create an account
+ * @static
+ * @description POST /api/v1/signUp
+ * @param {object} req request object
+ * @param {object} res ResponseService object
+ * @memberof UserController
+ * @returns {object} data
+ */
+  static async signUp(req, res) {
+    const userExist = await UserService.findByEmail(req.body.email.trim());
+    if (!userExist) {
+      const userInput = {
+        firstName: req.body.firstName.trim(),
+        lastName: req.body.lastName.trim(),
+        email: req.body.email.trim(),
+        password: BcryptService.hashPassword(req.body.password.trim())
+      };
+      const createdUser = await UserService.addUser(userInput);
+      const {
+        id, firstName, lastName, email, role, isVerified
+      } = createdUser;
+      const data = {
+        id, firstName, lastName, email, role, isVerified
+      };
+      ResponseService.setSuccess(201, 'User created successfully', data);
+      return ResponseService.send(res);
+    }
+    ResponseService.setError(409, `${req.body.email.trim()} already exist`);
+    return ResponseService.send(res);
+  }
+
+  /**
+   *
+   *User can log in
+   * @static
+   * @param {req} req
+   * @param {res} res
+   * @returns {token} @memberof UserController
+   */
+  static async login(req, res) {
+    const user = await UserService.findByEmail(req.body.email.trim());
```

Comment:
You should also check that the email address exist before verifying the password

Reply:
I am checking it in the middleware.